app/routes/$src.$id/tweets.tsx

```diff
@@ -110,6 +120,122 @@ export const loader: LoaderFunction = async ({ params, request }) => {
   return json<LoaderData>(tweets.map(wrapTweet), { headers });
 };
 
+interface BorgCluster {
+  active: boolean;
+  created_at: string;
+  id: string;
+  name: string;
+  updated_at: string;
+}
+
+interface BorgSocialAccount {
+  created_at: string;
+  description: string;
+  followers_count: string;
+  following_count: string;
+  id: string;
+  location: string;
+  name: string;
+  personal: boolean;
+  profile_image_url: string;
+  screen_name: string;
+  tweets_count: string;
+  updated_at: string;
+  url: string;
+}
+
+interface BorgInfluencer {
+  attention_score: number;
+  attention_score_change_week: number;
+  cluster_id: string;
+  created_at: string;
+  id: string;
+  identity: { clusters: BorgCluster[] };
+  insider_score: number;
+  personal_rank: string;
+  rank: string;
+  social_accounts: { social_account: BorgSocialAccount }[];
+  social_account: { social_account: BorgSocialAccount };
+}
+
+interface BorgResponse {
+  influencers: BorgInfluencer[];
+  total: string;
+  has_more: boolean;
+}
+
+async function getBorgCollectiveInfluencers(c: Cluster, pg = 0) {
+  log.debug(`Fetching influencers (${pg}) for ${c.name} (${c.id})...`);
+  const url =
+    `https://api.borg.id/influence/clusters/${c.name}/influencers?` +
+    `page=${pg}&sort_by=score&sort_direction=desc&influence_type=all`;
+  const headers = { authorization: `Token ${process.env.HIVE_TOKEN}` };
+  const data = (await (await fetch(url, { headers })).json()) as BorgResponse;
+  if (data.influencers && data.total) return data;
+  log.warn(`Fetched influencers: ${JSON.stringify(data, null, 2)}`);
+  return { influencers: [], total: 0 };
+}
+
+export const action: ActionFunction = async ({ params, request }) => {
+  const invocationId = nanoid(5);
+  console.time(`src-id-loader-${invocationId}`);
+  invariant(params.src, 'expected params.src');
+  invariant(params.id, 'expected params.id');
+  log.info(`Fetching tweets for ${params.src} (${params.id})...`);
+  const { api, session } = await getClient(request);
+  switch (params.src) {
```

Comment:
@ntorba I'm curious if you have any suggestions for how to organize these adapters (i.e. these different sources from which we determine what data to pull from Twitter): right now, we only have three (Hive/Borg Collective clusters, Twitter lists, and Rekt parlors) but in the future we'll want anyone to be able to easily add their own sources (e.g. you find a cool blogger who ranks systems design experts on their website and want to be able to see all the tweets from those design experts here, so you simply open a PR with a new adapter and we merge it). What should those adapters look like? WDYT?

Reply:
I think I answered my own question with all that reading I did yesterday.

We'll want to structure our Postgres database like [Mastodon](https://github.com/mastodon/mastodon) to use [ActivityPub](https://www.w3.org/TR/activitypub/) data types that are the "standard-ish" for DeSo apps. Then, to support Twitter (and any other existing centralized social platform like Instagram or LinkedIn), we'll simply create an adapter that consumes that centralized platform's public (or undocumented) API and spits out ActivityPub compatible data types (that are then stored in Postgres via another ActivityPub-to-Postgres adapter).